fakestorage/object.go

```diff
@@ -234,7 +234,7 @@ func (s *Server) createObject(obj Object) (Object, error) {
 
 // ListObjects returns a sorted list of objects that match the given criteria,
 // or an error if the bucket doesn't exist.
-func (s *Server) ListObjects(bucketName, prefix, delimiter string, versions bool) ([]Object, []string, error) {
+func (s *Server) ListObjects(bucketName, prefix, delimiter, startOffset, endOffset string, versions bool) ([]Object, []string, error) {
```

Comment:
fake-gcs-server is also used as a library and this would be a breaking change. Can we introduce the feature without modifying the public API? Maybe create a method called `retrieveObjectsList` that takes all parameters you have currently and keep the signature of `ListObjects` the same (and `ListObjects` would just call `s.listObjects(bucketName, prefix, delimiter, "", "", versions)`). Or since the list of parameters is too long anyways, we could introduce a `ListObjectsWithOptions` that takes all the parameters as a struct. That would allow us to continue to evolve the API.

What do you think?

Reply:
Hmmm, sorry, I didn't realize it was also used as a library. Anyway, I really like the idea of adding a new `ListObjectsWithOptions(bucketName, options)`, so I'll do that instead. Thanks for the suggestion!

Reply:
Should be good now. Let me know what you think